Lib/pdb.py

```diff
@@ -422,15 +424,28 @@ def interaction(self, frame, tb_or_exc):
         _exceptions = []
         if isinstance(tb_or_exc, BaseException):
             traceback, exception = tb_or_exc.__traceback__, tb_or_exc
-            current = exception
-            while current:
+            currents = [exception]
+
+            while currents:
+                current = currents.pop()
+                if current in _exceptions:
+                    continue
```

Comment:
If we put a limit of 999, maybe we shouldn't do anything special for cycles after all, just let them reach the limit?

Reply:
Well if we have just A->B->C->A, it's still better to show "only" 3 exceptions than 999 ? 

I'm guessing the 999 limit might be more relevant for exception groups in async framework where you may have 1000+ tasks failing ?